### PR TITLE
VFB-5 Data Input Components

### PR DESCRIPTION
**Pull request by daniel-conlan-softwire**
Implemented a new data input component `CheckboxGroupInput` alongside a corresponding change handler factory function in `components/DataInput/inputHandlerFactories`. 

The checkbox group is a set of checkbox form elements and is used when the state of each checkbox within the group are related to one another. For example, a group of checkboxes used to select the toppings on a pizza that are not mutually exclusive to one another.

Appropriate tests have also been provided.

- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've linted via `npm run lint`
    - can run `npm run lint_fix` to try and fix as any mistakes automatically as possible!
- [x] Make sure you've tested via `npm run test`

### CODE REVIEW

**Review by Var1377 — APPROVED** [review]

**Review by SamYap-Softwire — APPROVED** [review]

File: src/components/DataInput/CheckboxGroupInput.tsx
Comment (TomCusack-Softwire) [inline src/components/DataInput/CheckboxGroupInput.tsx:9]: should be able to remove this disable!

File: src/components/DataInput/CheckboxGroupInput.tsx
Comment (TomCusack-Softwire) [inline src/components/DataInput/CheckboxGroupInput.tsx:15]: if `props.groupLabel` is undefined, should this be label with empty string or no label at all?

**Review by TomCusack-Softwire — COMMENTED** [review]
From the abundance of commits, it looks like you reused the previous VFB5 branch? 
try not to do this - ideally you delete the branch as soon as it's merged and create a new branch for each PR!

also PR title should have an imperative verb which fits the sentence 'this commit will _'
e.g. `VFB-123 Add a row to a table`

**Review by TomCusack-Softwire — APPROVED** [review]
I'm happy to merge once the above changes are implemented!

File: src/components/DataInput/CheckboxGroupInput.tsx
Comment (daniel-conlan-softwire) [reply src/components/DataInput/CheckboxGroupInput.tsx:15]: If `props.groupLabel` is `undefined` then the label just isn't rendered. It's all handled by the MUI component

**Pull request merged**
